src/modules/claim/services/claim-verification.service.ts

```diff
@@ -0,0 +1,149 @@
+import { Injectable } from '@nestjs/common';
+import { Logger } from '../../logger/logger.service';
+import { IssuerVerificationService } from './issuer-verification.service';
+import { DIDService } from '../../did/did.service';
+import { RoleCredentialResolver } from '../resolvers/credential.resolver';
+import { ProofVerifier } from '@ew-did-registry/claims';
+import {
+  CredentialResolver,
+  RoleEIP191JWT,
+  isEIP191Jwt,
+} from '@energyweb/vc-verification';
+
+@Injectable()
+export class ClaimVerificationService {
+  credentialResolver: CredentialResolver;
+  constructor(
+    private readonly logger: Logger,
+    private readonly didService: DIDService,
+    private readonly issuerVerificationService: IssuerVerificationService
+  ) {
+    this.logger.setContext(ClaimVerificationService.name);
+    this.credentialResolver = new RoleCredentialResolver(didService);
+  }
+
+  /**
+   * Resolve a credential from storage and verify its proof/signature and its issuer's authority
+   *
+   * @param subjectDID The DID to try to resolve a credential for
+   * @param roleNamesapce The role to try to get a credential for. Should be a full role namespace (for example, "myrole.roles.myorg.auth.ewc")
+   * @return void. Returns boolean indicating if credential is verified. Contains array of error messages if not verified.
+   */
+  public async resolveCredentialAndVerify(
```

Comment:
@jrhender Should I be adding a test suite for this file for each of these methods?

Reply:
I went ahead and did tests for these methods.